File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]
        
        for s in strs[1:]:
            
            i, j = 0, 0
            while i < len(prefix) and j < len(s):
                if prefix[i] != s[j]:
                    break
                
                i += 1
                j += 1
            
            prefix = prefix[:i]
        
        return prefix

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    var prefix = strs[0];
    
    for (let i = 1; i < strs.length; i++){
        while (strs[i].slice(0, prefix.length) != prefix){
            prefix = prefix.slice(0, prefix.length-1);
            
            if (! prefix) return "";
        }
    }
    
    return prefix;
};